agent/engine/dockerstate/json.go

```diff
@@ -55,47 +55,50 @@ func (state *DockerTaskEngineState) UnmarshalJSON(data []byte) error {
 	if err != nil {
 		return err
 	}
-	// reset it by just creating a new one and swapping shortly.
-	// This also means we don't have to lock for the remainder of this function
-	// because we are the only ones with a reference to clean
-	clean := newDockerTaskEngineState()
 
+	// run precheck; shake out all the errors before resetting state.
```

Comment:
why do we need to precheck?

Reply:
don't want any of those errors to happen while we're setting up the state, leaving it in a nasty... well... state.